types/testing.go

```diff
@@ -65,6 +76,44 @@ func (ms MockSigner) SignBytes(data []byte, addr address.Address) (Signature, er
 	return crypto.Sign(ki.Key(), hash[:])
 }
 
+// AddressFromPubKey looks up a KeyInfo address associated with a given PublicKey for a MockSigner
+func (ms MockSigner) AddressFromPubKey(pk []byte) (address.Address, error) {
+	var addr address.Address
+
+	for _, ki := range ms.AddrKeyInfo {
+		testPk := ki.PublicKey()
+
+		if bytes.Equal(testPk, pk) {
+			addr, err := ki.Address()
+			if err != nil {
+				return addr, errors.New("could not fetch address")
+			}
+			return addr, nil
+		}
+	}
+	return addr, errors.New("public key not found in wallet")
+}
+
+// CreateTicket is effectively a duplicate of Wallet CreateTicket for testing purposes.
```

Comment:
Since this code seems to exercise the actual signing calls why not use a wallet in tests?

Reply:
Because we are using a mockSigner in many tests where we don't always have access to, or want to go through the hassle of creating a wallet.  I tried just using Wallet and I wound up with massive issues, import conflicts being the least of it.

Reply:
@ZenGround0 it turns out with the requested refactor to put CreateTicket back into consensus, the interface implementation means we actually probably _can_ use a wallet.  But this PR is too long so I filed issue #2252 and will follow up with it.